presto-main/src/main/java/com/facebook/presto/cost/SimpleStatsRule.java

```diff
@@ -0,0 +1,45 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.cost;
+
+import com.facebook.presto.Session;
+import com.facebook.presto.spi.type.Type;
+import com.facebook.presto.sql.planner.Symbol;
+import com.facebook.presto.sql.planner.iterative.Lookup;
+import com.facebook.presto.sql.planner.plan.PlanNode;
+
+import java.util.Map;
+import java.util.Optional;
+
+import static java.util.Objects.requireNonNull;
+
+public abstract class SimpleStatsRule
+        implements ComposableStatsCalculator.Rule
+{
+    private final StatsNormalizer normalizer;
+
+    protected SimpleStatsRule(StatsNormalizer normalizer)
+    {
+        this.normalizer = requireNonNull(normalizer, "normalizer is null");
```

Comment:
do stats rule need to share the normalizer? Why not to just create a normalizer here.

Reply:
Why not?